crates/oxc_transformer/src/typescript/module.rs

```diff
@@ -104,7 +104,15 @@ impl<'a> TypeScriptModule<'a, '_> {
                     self.ctx.error(diagnostics::import_equals_cannot_be_used_in_esm(decl_span));
                 }
 
-                let callee = ctx.ast.expression_identifier_reference(SPAN, "require");
+                // `TSImportEqualsDeclaration` can only appear at top level, so we only need to check
+                // root scope for `require` binding
+                let require_symbol_id = ctx.scopes().get_root_binding("require");
+                let callee = ctx.create_ident_expr(
+                    SPAN,
+                    Atom::from("require"),
+                    require_symbol_id,
+                    ReferenceFlags::Read,
+                );
```

Comment:
Is my assumption that `TSImportEqualsDeclaration` is always at top level here correct? Or is something like this maybe legal?

```ts
import {LongNameModule} from 'foo';
module foo {
  import AliasModule = LongNameModule;
}
```

Reply:
`TSImportEqualsDeclaration` can work in `Namespace` 

https://www.typescriptlang.org/play/?target=99#code/JYWwDg9gTgLgBAbwDIQHYHMByBDEBTAWQgBMBXAGzwF84AzKCEOAclogmYG44AoEEinjrtEPOONCRYcAILlg2AM5EylOAF44KDDnwrBnMeIDGaRREoA6chHQAKOQuUDKASiPiqQA

Reply:
Fixed. Now using current scope as starting point for searching for `require` binding.

But... #8384!